app/views.py

```diff
@@ -151,3 +156,291 @@ def not_found(error):
 @app.errorhandler(500)
 def server_error(error):
     return render_template("server_error.html"), 500
+
+
+@app.route("/", methods=["GET", "POST"])
+def home():
+    if current_user.is_authenticated:
+        query = request.args.get("query")
+        categories = Category.query.filter_by(user_id=current_user.id).all()
+        category_id = request.args.get("category_id")
+        if query:
+            notes = (
+                Note.query.options(joinedload(Note.category))
+                .filter_by(user_id=current_user.id)
+                .filter(Note.name.ilike(f"%{query}%"))
+            )
+        elif category_id:
+            notes = (
+                Note.query.options(joinedload(Note.category))
+                .filter_by(user_id=current_user.id)
+                .filter_by(category_id=category_id)
+                .all()
+            )
+        else:
+            notes = (
+                Note.query.options(joinedload(Note.category))
+                .filter_by(user_id=current_user.id)
+                .all()
+            )
+        note_images = {}
+        for note in notes:
+            images = (
+                Image.query.filter_by(note_id=note.id).order_by(Image.id.asc()).all()
+            )
+            note_images[note.id] = images
+        return render_template(
+            "index.html",
+            notes=notes,
+            note_images=note_images,
+            os=os,
+            categories=categories,
+        )
+    return render_template("index.html")
+
+
+@app.route("/search-notes")
+@login_required
+def search_notes():
+    categories = Category.query.filter_by(user_id=current_user.id).all()
+    query = request.args.get("query")
+    notes = (
+        Note.query.options(joinedload(Note.category))
+        .filter_by(user_id=current_user.id)
+        .filter(Note.name.ilike(f"%{query}%"))
+        .all()
+    )
+    note_images = {}
+    for note in notes:
+        images = Image.query.filter_by(note_id=note.id).order_by(Image.id.asc()).all()
+        note_images[note.id] = images
+    return render_template(
+        "index.html", notes=notes, note_images=note_images, os=os, categories=categories
+    )
+
+
+@app.route("/edit_note/<int:note_id>", methods=["GET", "POST"])
```

Comment:
Kadangi view nėra logikos, kuri patikrintų ar vartotojas gali redaguoti šitą įrašą, tai bet kuris vartotojas gali pasiekti ir redaguoti kitų vartotojų įrašus.